libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Recognizers;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+using Microsoft.Orchestrator;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.AI.Orchestrator
+{
+    public class OrchestratorAdaptiveRecognizer : Recognizer
```

Comment:
>OrchestratorAdaptiveRecognizer [](start = 17, length = 30)

I would rename this OrchestratorRecognizer.  The only reason we had "LuisAdaptiveRecognizer" was because we had a legacy LuisRecognizer that we couldn't change.

Reply:
Done. Removed OrchestratorRecognizer. Renamed OrchestratorAdaptiveRecognizer to OrchestratorRecognizer. OrchestratorRecognizer class now serves the need of both dispatch and adaptive use cases.

Reply:
The `OrchestratorRecognizer` class still exists in .NET (f2acd8d) and JS. Both the `OrchestratorAdaptiveRecognizer` and `OrchestratorRecognizer` classes are public, but the non-Adaptive version has a dependency on the Adaptive `Recognizer`. Do we actually need both classes?

The OrchestratorRecognizer also doesn't serve the needs of both dispatch and adaptive in JS, because the original `dispatch` in a browser-based scenario is performed by LUIS, which isn't reliant on a filesystem. Orchestrator and the declarative library heavily rely on the filesystem which points toward a need to be explicit about what is and isn't supported via the JS library.

Reply:
Yes @stevengum there was a need to keep both and I chatted with @tomlm about this. We need both classes because we want to keep use of orchestratorRecognizer be a pattern that is aligned with how we talk about [using dispatch today][1]. We do the same thing for LUISrecognizer and LuisAdaptiveRecognizer.

For adaptive in both C# and JS, the customer would use OrchestratorAdaptiveRecognizer (similar to LuisAdaptiveRecognizer). For dispatch scenario, the customer would use OrchestratorRecognizer (similar to LuisRecognizer). 

I'm not sure how the dependency on the file system relates to dispatch.

Reply:
Besides configuration loading, are there behavioral changes between LuisRecognizer and LuisAdaptiveRecognizer? 

> For adaptive in both C# and JS, the customer would use OrchestratorAdaptiveRecognizer (similar to LuisAdaptiveRecognizer). For dispatch scenario, the customer would use OrchestratorRecognizer (similar to LuisRecognizer).

Implicitly speaking, it sounds like there are differences between the Adaptive-scenario and the dispatch-scenario for Orchestrator... Is this correct?

Reply:
There are no functional differences between them.